Arriba/Arriba/Model/Partition.cs

```diff
@@ -443,6 +452,9 @@ public void FillPartitionColumn(Partition p, DataBlock values, int columnIndex,
 
                 for (int rowIndex = 0, sourceItemIndex = chainStartIndex; rowIndex < rowCount; ++rowIndex, sourceItemIndex = partitionChains[sourceItemIndex])
                 {
+                    // If the item is new and no LID was assigned, we don't set values
```

Comment:
Can you expand the comment here to explain how this could happen.  Absent understanding the above loop this seems like an odd/error condition.